src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -84,6 +87,45 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
     Await.result(futureStore, DatabaseTimeout)
   }
 
+ /**
+   * Try to generate output for a collector call, by collecting outputs for all of its shards.
+   * It's fail-fast on shard output retrieval
+   */
+  def generateCollectorOutput(collector: CollectorKey, shards: Iterable[CallKey]): Try[CallOutputs] = Try {
+    val shardsOutputs = shards map { e =>
+      fetchCallOutputEntries(e.scope) map { _.value } get
+    }
+   /* FIXME: No need for asInstanceOf[Call] when merged to "Database SG branch" */
+    collector.scope.asInstanceOf[Call].task.outputs map { taskOutput =>
+      val wdlValues = shardsOutputs.map(_.get(taskOutput.name).get).toSeq
+      taskOutput.name -> new WdlArray(WdlArrayType(WdlAnyType), wdlValues)
+    } toMap
+  }
+
+  def findCollectorChildren(key: CollectorKey) = executionStore collect {
+    case (k: CallKey, _) if k.scope == key.scope && isShard(k) => k
+  }
+
+  /**
+   * Gathers and completes Collector Calls.
+   */
+  def collectScatter(collector: CollectorKey) = {
+    val shards: Iterable[CallKey] = findCollectorChildren(collector)
```

Comment:
if "collector children" == "shard" maybe the method should be `findShards(collector)`?